### PR TITLE
Updates for golang pr

### PR DESCRIPTION
**Pull request by vsoch**
This will be a pull request (in sync with one to cmisale/flux-sched) that will make changes here necessary to work with the new structure, specifically returning errors instead of ints. I also ran gofmt in a separate commit to clean up some of the formatting. I will alert folks with one message (and links to both PRs) in chat when this is done.

Note that since these are all tied together, to test I am needing to update the module install to use the commit of my WIP branches. We will need to do some final update/merge to point them all to flux-framework repos (suggested steps below).
Here is my suggestion for order of operations:

1. Superficial review of here and https://github.com/cmisale/flux-sched/pull/6.
2. When https://github.com/cmisale/flux-sched/pull/6 is good, it gets merged into https://github.com/flux-framework/flux-k8s/pull/41
3. We then rename the module name there from researchapps -> flux-framework and merge
4. Then we come back here, and I can update the path accordingly, and we should be able to merge.

To test (aside from tests alongside flux-sched) - I went through the [Deploy Fluence](https://github.com/flux-framework/flux-k8s#building-fluence) steps and did the build of the images. However, before the first `make` for the sidecar container, the go.mod needs to be changed with the latest commit of the flux-sched testing branch (which might be a moving target). E.g., before running the first `make` I did (this is the current commit as of writing this):

```bash
cd ./scheduler-plugin/fluence
go get github.com/researchapps/flux-sched/resource/reapi/bindings/go/src/fluxcli@8789094b6ee71e9d00e7b21785c2ad5e04de342e

# Updated for August 23
go get github.com/researchapps/flux-sched/resource/reapi/bindings/go/src/fluxcli@9089091d08f86a3bb6b228fff4ffbdcb6ad413c3
go mod tidy

# Back to root with Makefile that will build the Dockerfile in build/scheduler/Dockerfile
make
```
And then continue following instructions to push to registries and test. I went through this whole process to test the changes here, and then checked that tests pass, and that's the only two things I know to test!

### CODE REVIEW

File: scheduler-plugin/fluence/cmd/main.go [hunk 1]
```diff
@@ -1,25 +1,24 @@
 package main
 
 import (
-	"fmt"
 	"flag"
-	"net"
-	"google.golang.org/grpc/keepalive"
+	"fmt"
```
Comment (milroy) [inline scheduler-plugin/fluence/cmd/main.go:5]: Does the import order matter?

**Review by milroy — APPROVED** [review]
This looks good. Thanks a lot!

**Comment by milroy** [comment]
> Superficial review of here and https://github.com/cmisale/flux-sched/pull/6.
> When https://github.com/cmisale/flux-sched/pull/6 is good, it gets merged into https://github.com/flux-framework/flux-k8s/pull/41
> We then rename the module name there from researchapps -> flux-framework and merge
> Then we come back here, and I can update the path accordingly, and we should be able to merge.

This makes sense to me.

File: scheduler-plugin/fluence/cmd/main.go (hunk 1 above)
Comment (vsoch) [reply scheduler-plugin/fluence/cmd/main.go:5]: They do - I usually let my editor (VSCode) determine the order (details here https://teamtreehouse.com/community/does-it-matter-what-order-imports-go) and if there is something that breaks based on that convention I’d expect the library not to work (although this hasn’t happened to me yet!)

**Comment by milroy** [comment]
I will add that adding a short description of the commit in each commit body.

**Comment by vsoch** [comment]
Also note I don’t think we should merge here yet (see order of operations in description bullets) and feel free to critique that too.

**Comment by vsoch** [comment]
> I will add that adding a short description of the commit in each commit body.

Thank you! :pray:

**Comment by milroy** [comment]
Actually, I do have a question about the order of operations. For item 2:

> When https://github.com/cmisale/flux-sched/pull/6 is good, it gets merged into https://github.com/flux-framework/flux-k8s/pull/41

Do you mean When https://github.com/cmisale/flux-sched/pull/6 is good, it gets merged in as https://github.com/flux-framework/flux-sched/pull/1008?

**Comment by vsoch** [comment]
Let me try to remember (and better articulate).

1. Merging https://github.com/cmisale/flux-sched/pull/6 immediately becomes part of https://github.com/flux-framework/flux-sched/pull/1008 because it's the same branch golang-pr.
2. When that merge is done, we need to update references of "researchapps" to "flux-framework" ([here](https://github.com/cmisale/flux-sched/pull/6/files#diff-5639eddccc6d4d09fd13cd7563b21acc9f7adc4904e115047a85e6d96444d75aR2) is an example)
3. Then we come back here, and we work again on changes here, because there are references to cmisale/flux-sched scattered around. We need flux-framework [here](https://github.com/flux-framework/flux-k8s/pull/41/files#diff-9bd1dc515b572a36e3959146fe0ef7aee4ffcf81ce634483d915f5dcda470162R9) and [here](https://github.com/flux-framework/flux-k8s/blob/01213c29e99183ffababb0bf84ed801506f248bd/scheduler-plugin/build/scheduler/Dockerfile#L90-L97) (and probably a few other places too).

I had thought there was a circular relationship somewhere but I can't find it now, so maybe it's not as complicated as I thought. If there is something I am missing, I promise to take responsibility for coming back and fixing anything that is an oversight on my part.

**Comment by milroy** [comment]
> If there is something I am missing, I promise to take responsibility for coming back and fixing anything that is an oversight on my part.

I know you would, and this isn't an oversight! It's just something that confused me when I reread it.

**Pull request closed without merging**